src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c

```diff
@@ -21,69 +63,1180 @@ const ActorInit Boss_Fd2_InitVars = {
+        Math_SmoothScaleMaxF(&this->unk_13A0.x, this->unk_13F4.x, this->unk_1400.x, this->unk_13D0.x * this->unk_140C);
+        Math_SmoothScaleMaxF(&this->unk_13A0.y, this->unk_13F4.y, this->unk_1400.y, this->unk_13D0.y * this->unk_140C);
+        Math_SmoothScaleMaxF(&this->unk_13A0.z, this->unk_13F4.z, this->unk_1400.z, this->unk_13D0.z * this->unk_140C);
+        Math_SmoothScaleMaxF(&this->unk_140C, 1.0f, 1.0f, this->unk_1410);
+        this->unk_13A0.y += this->unk_1414;
+        func_800C04D8(globalCtx, this->cutsceneCamera, &this->unk_13A0, &this->unk_1394);
+        Math_SmoothScaleMaxF(&this->unk_1414, 0.0f, 1.0f, 0.1f);
+    }
+}
+
+void BossFd2_Death(BossFd2* this, GlobalContext* globalCtx) {
+    f32 sp7C;
+    Vec3f sp70;
+    Vec3f sp64;
+    BossFd* bossFd = BOSSFD;
+    Camera* camera = Gameplay_GetCamera(globalCtx, 0);
+    Vec3f* temp_eye;
+    Vec3f* temp_at;
+    SkelAnime* skelAnime = &this->skelAnime;
+    f32 temp_f0;
+    s32 pad1;
+
+    SkelAnime_FrameUpdateMatrix(skelAnime);
+    switch (this->cutsceneState) {
+        case 0:
+            this->cutsceneState = 1;
+            func_80064520(globalCtx, &globalCtx->csCtx);
+            func_8002DF54(globalCtx, &this->actor, 1);
+            this->cutsceneCamera = Gameplay_CreateSubCamera(globalCtx);
+            Gameplay_ChangeCameraStatus(globalCtx, 0, 1);
+            Gameplay_ChangeCameraStatus(globalCtx, this->cutsceneCamera, 7);
+            this->unk_1394 = camera->eye;
+            this->unk_13A0 = camera->at;
+            this->unk_13C4.x = 100.0f;
+            this->unk_13C4.y = 100.0f;
+            this->unk_13C4.z = 100.0f;
+            this->unk_13D0.x = 100.0f;
+            this->unk_13D0.y = 100.0f;
+            this->unk_13D0.z = 100.0f;
+            this->unk_1410 = 0.02f;
+            this->timers[0] = 0;
+            this->holeCounter = 0;
+            this->unk_13E8.x = 0.1f;
+            this->unk_13E8.y = 0.1f;
+            this->unk_13E8.z = 0.1f;
+            this->unk_1400.x = 0.1f;
+            this->unk_1400.y = 0.1f;
+            this->unk_1400.z = 0.1f;
+        case 1:
+            this->holeCounter++;
+            if (this->holeCounter < 15) {
+                sp7C = 1.0f;
+            } else if (this->holeCounter < 20) {
+                sp7C = 0.5f;
+            } else {
+                sp7C = 0.25f;
+            }
+            if ((this->holeCounter == 1) || (this->holeCounter == 40)) {
+                this->screamTimer = 20;
+                if (this->holeCounter == 40) {
+                    func_800F8D04(NA_SE_EN_VALVAISA_DEAD);
+                }
+
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_VALVAISA_DAMAGE2);
+            }
+            Math_SmoothScaleMaxF(&this->skelAnime.animPlaybackSpeed, sp7C, 1.0f, 1.0f);
+            Matrix_RotateY(((this->actor.yawTowardsLink / 32768.0f) * M_PI) + 0.2f, 0);
```

Comment:
I noticed you had some `32768.0f` as `(f32)0x8000`, maybe use that everywhere?
Do you have to cast if you do eg `this->actor.yawTowardsLink * (M_PI / 0x8000)` ?

Reply:
cast is needed because it's two separate operations. I tried it in some other things and it doesn't always match, but I might want to try again because maybe I was being tripped up by the deduplicator in unclear ways. Float match is super delicate though.

Reply:
tried it again and it works. Must have done it wrong last time.